tests/options/sortAttributes/sort-attributes.test.ts

```diff
@@ -38,19 +39,181 @@ describe('Options', () => {
+		});
+		test('compare 4', () => {
+			const pugSortAttributes: SortAttributes = 'asc';
+			const pugSortAttributesBeginning: string[] = ['^x$', '^y$', '^z$'];
+			const pugSortAttributesEnd: string[] = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
+			const expected: ReadonlyArray<string> = [
+				'x',
+				'y',
+				'z',
+				'D',
+				'a',
+				'b',
+				'c',
+				'v-for',
+				':key',
+				'src',
+				'alt',
+				'@click',
+				':disabled'
+			];
+			const code: string[] = [
+				'y',
+				'c',
+				'z',
+				'a',
+				':disabled',
+				'alt',
+				'b',
+				'D',
+				':key',
+				'v-for',
+				'@click',
+				'src',
+				'x'
+			];
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 5', () => {
+			const pugSortAttributes: SortAttributes = 'desc';
+			const pugSortAttributesBeginning: string[] = ['^x$', '^y$', '^z$'];
+			const pugSortAttributesEnd: string[] = ['v-for', ':key', 'src', 'alt', '@click', ':disabled'];
+			const expected: ReadonlyArray<string> = [
+				'x',
+				'y',
+				'z',
+				'c',
+				'b',
+				'a',
+				'v-for',
+				':key',
+				'src',
+				'alt',
+				'@click',
+				':disabled'
+			];
+			const code: string[] = ['y', 'c', 'z', 'a', ':disabled', 'alt', 'b', ':key', 'v-for', '@click', 'src', 'x'];
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 6', () => {
+			const pugSortAttributes: SortAttributes = 'as-is';
+			const pugSortAttributesBeginning: string[] = ['a'];
+			const pugSortAttributesEnd: string[] = ['b'];
+			const expected: ReadonlyArray<string> = 'aedcfghilnb'.split('');
+			const code: string[] = 'aedcfbghiln'.split('');
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
+			);
+
+			expect(actual).toStrictEqual(expected);
+		});
+		test('compare 7', () => {
+			const pugSortAttributes: SortAttributes = 'as-is';
+			const pugSortAttributesBeginning: string[] = [];
+			const pugSortAttributesEnd: string[] = [];
+			const expected: ReadonlyArray<string> = 'aedcfghilnb'.split('');
+			const code: string[] = 'aedcfghilnb'.split('');
+			const actual: string[] = stableSort(code, (a, b) =>
+				compareAttributeToken(
+					createAttributeToken(a),
+					createAttributeToken(b),
+					pugSortAttributes,
+					pugSortAttributesBeginning,
+					pugSortAttributesEnd
+				)
```

Comment:
Can you add a test using beginning and end but don't pass `pugSortAttributes` so it takes the default one?